src/runtime/fake_thread_pool.cpp

```diff
@@ -90,6 +90,7 @@ struct halide_mutex_array {
 };
 
 WEAK halide_mutex_array *halide_mutex_array_create(int sz) {
+    halide_error(nullptr, "halide_mutex_array_create not implemented on this platform.");
```

Comment:
Should this just return a non-null dummy pointer? Maybe a pointer to a static/global default constructed halide_mutex_array? I'm assuming the design is that halide_mutex_array is an opaque type outside of this file, and all of the implementation functions for it in this file just do nothing.

If so, the rest of the changes in this PR could probably be reverted.

Reply:
Hm, that's an interesting thought. I guess having the same pointer returned for multiple calls is fine if you assume that external code doesn't care about the value (and that the other calls are all no-ops). The only point of concern I would have is that we'd have to ensure the semantics of the non-fake-thread versions don't grow a use that is incompatible with this.